Test/MergeFile.Tests.ps1

```diff
@@ -0,0 +1,303 @@
+    It 'should delete source files' {
+        Init
+        GivenFile 'one.txt' -WithContent 'one'
+        GivenFile 'two.txt' -WithContent 'two'
+        WhenMerging 'one.txt','two.txt' -Into 'merged.txt' -AndDeletingSourceFiles
+        ThenFile 'one.txt' -DoesNotExist
+        ThenFile 'two.txt' -DoesNotExist
+        ThenFile 'merged.txt' -HasContent 'onetwo'
+    }
+}
+
+Describe 'MergeFile.when customizing separator' {
+    It 'should separate file contents with separator' {
+        Init
+        GivenFile 'one.txt' -WithContent 'one'
+        GivenFile 'two.txt' -WithContent 'two'
+        WhenMerging 'one.txt','two.txt' -Into 'merged.txt' -WithSeparator ('$(NewLine)') 
+        ThenFile 'merged.txt' -HasContent ('one{0}two' -f [Environment]::NewLine)
+    }
+}
+
+Describe 'MergeFile.when customizing separator and keeping destination file content' {
+    It 'should add separator after existing contents' {
+        Init
+        GivenFile 'one.txt' -WithContent 'one'
+        GivenFile 'two.txt' -WithContent 'two'
+        GivenFile 'merged.txt' -WithContent 'i was here first'
+        WhenMerging 'one.txt','two.txt' -Into 'merged.txt' -WithSeparator ('$(NewLine)') 
+        ThenFile 'merged.txt' -HasContent ('i was here first{0}one{0}two' -f [Environment]::NewLine)
+    }
+}
+
+Describe 'MergeFile.when customizing separator and clearing destination file content' {
+    It 'should not add separator to beginning of file' {
+        Init
+        GivenFile 'one.txt' -WithContent 'one'
+        GivenFile 'two.txt' -WithContent 'two'
+        GivenFile 'merge.tst' -WithContent 'gone gone gone'
+        WhenMerging 'one.txt','two.txt' -Into 'merged.txt' -WithSeparator ('$(NewLine)') -Clear
+        ThenFile 'merged.txt' -HasContent ('one{0}two' -f [Environment]::NewLine)
+    }
+}
+
+Describe 'MergeFile.when files are empty' {
+    It 'should separate file contents with separator' {
+        Init
+        GivenFile 'one.txt' 
+        GivenFile 'two.txt' 
+        WhenMerging 'one.txt','two.txt' -Into 'merged.txt' 
+        ThenFile 'merged.txt' -HasContent ('')
+    }
+}
+
+Describe 'MergeFile.when files are binary' {
+    It 'should concatenate files' {
+        Init
+        [byte[]]$content = Get-RandomByte
+        GivenFile 'one.txt' -WithContent $content 
+        [byte[]]$content2 = Get-RandomByte
+        GivenFile 'two.txt' -WithContent ($content2)
+        WhenMerging 'one.txt','two.txt' -Into 'merged.txt'
+        ThenFile 'merged.txt' -HasContent ([byte[]]($content + $content2))
+    }
+}
+
+Describe 'MergeFile.when separator and files are binary' {
+    It 'should concatenate files with separator' {
+        Init
+        [byte[]]$content = Get-RandomByte
+        GivenFile 'one.txt' -WithContent $content 
+        [byte[]]$content2 = Get-RandomByte
+        GivenFile 'two.txt' -WithContent ($content2)
+        WhenMerging 'one.txt','two.txt' -Into 'merged.txt' -WithSeparator ([byte[]]@( 28 ))

```

Comment:
We should have a test to ensure that the `TextSeparator` property takes precedence over the `BinarySeparator` property.

Reply:
I added a failure condition if both are provided.